### PR TITLE
docs: Orval ADR

### PR DESCRIPTION
**Pull request by Tymek**
## About the changes
Documenting the decision.

Closes [1-537/openapi-generator-adr](https://linear.app/unleash/issue/1-537/openapi-generator-adr)

### CODE REVIEW

File: website/docs/contributing/ADRs/front-end/sdk-generator.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+title: "ADR: OpenAPI SDK Generator"
+---
+
+## Background
+
+Our current frontend setup we have a lot of code that can be generated out of OpenAPI schema. Types have not been updated in a while, and even some new features rely on hard-coded types instead of auto-generated files in `src/openapi`. Fetchers and actions in the frontend involves custom-built code that is grouped in the `src/hooks/api` folder. There is a separation between getters and actions. Getters use the SWR library. API actions (POST/PUT/DELETE) are grouped by feature and are exposed to components reliant on `useAPI` hook.
```
Comment (kwasniew) [inline website/docs/contributing/ADRs/front-end/sdk-generator.md:7]: In our current ...

Comment (kwasniew) [reply website/docs/contributing/ADRs/front-end/sdk-generator.md:7]: of the OpenAPI schema

Comment (kwasniew) [reply website/docs/contributing/ADRs/front-end/sdk-generator.md:7]: in the frontend involve

File: website/docs/contributing/ADRs/front-end/sdk-generator.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+SDK generated out of it will be better then what we have right now. It will help to reduce the risk of duplication and inconsistencies in our SDK, as it will be generated from a single source of truth.
```
Comment (kwasniew) [inline website/docs/contributing/ADRs/front-end/sdk-generator.md:9]: This is the advantage, not background

File: website/docs/contributing/ADRs/front-end/sdk-generator.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+- We will use the [Orval](https://orval.dev/) package to generate the typescript types for our SDK.
+- We will consider using Orval to generate the HTTP getters and actions for our SDK in the future, but will first carefully test this approach in new features under development to weed out edge cases.
+- We will deprecate `src/interfaces` related to API calls and use `src/openapi` models instead.
+
+### Advantages
+
+- We retain the flexibility of previous solution, because we can implement our own _fetcher_ function, and substitute _response_ and _error_ type generics. See https://orval.dev/guides/custom-client
+- Orval supports `anyOf` and `oneOf` schema, which the previous generator did not support.
+- If we decide to use Orval to generate the HTTP getters and actions for our SDK, it will reduce the amount of boilerplate code required when working with new APIs.
```
Comment (kwasniew) [inline website/docs/contributing/ADRs/front-end/sdk-generator.md:21]: the new APIs

File: website/docs/contributing/frontend/overview.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ The frontend is written in react/typescript. It's is a single page application t
 
 We have created a set of ADRs to help guide the development of the frontend:
 
+* [OpenAPI SDK generator](../ADRs/front-end/sdk-generator.md)
```
Comment (kwasniew) [inline website/docs/contributing/frontend/overview.md:13]: why is it added as the first one?

File: website/docs/contributing/ADRs/front-end/sdk-generator.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- We can always revert to writing some API calls by hand if this approach is not flexible enough.
+
+### Concerns
```
Comment (kwasniew) [inline website/docs/contributing/ADRs/front-end/sdk-generator.md:24]: In terms of concerns this twitter conversation summarizes it nicely: https://twitter.com/gicagalbenu/status/1494933892310212611

**Review by kwasniew — APPROVED** [review]
Non blocking approval but please check the comments

File: website/docs/contributing/ADRs/front-end/sdk-generator.md (hunk 2 above)
Comment (nunogois) [inline website/docs/contributing/ADRs/front-end/sdk-generator.md:9]:
```suggestion
An SDK generated out of it will be better than what we have right now. It will help reduce the risk of duplication and inconsistencies in our SDK, as it will be generated from a single source of truth.
```

File: website/docs/contributing/ADRs/front-end/sdk-generator.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+- We will need to ensure that we keep our OpenAPI specification up-to-date, as any changes in the specification will be reflected in the generated SDK. We need an enterprise version with all experimental endpoints enabled to get complete output.
+- Orval is well-maintained, but it appears to have just 1 core contributor. SDK is a very important thing and should be reliable.
+
+## Alternative packages considered
+
+- `@openapitools/openapi-generator` - it because it does not offer as many customization options as Orval. It struggles with `anyOf` and `oneOf` types. It fails
```
Comment (nunogois) [inline website/docs/contributing/ADRs/front-end/sdk-generator.md:31]:
```suggestion
- `@openapitools/openapi-generator` - Does not offer as many customization options as Orval. It struggles with `anyOf` and `oneOf` types. It fails 
```
